### PR TITLE
Cross service storages: Simplify PRs

### PR DESCRIPTION
**Pull request by netopyr**
This change gathers all required changes in the dispatcher in a single PR to avoid merge conflicts in the other PRs.

### CODE REVIEW

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/dispatcher/TransactionDispatcher.java [hunk 1]
```diff
@@ -116,86 +116,81 @@ public void dispatchHandle(
...
-            }
+            case SYSTEM_DELETE -> switch (txBody.systemDeleteOrThrow().id().kind()) {
+                case CONTRACT_ID -> handlers.contractSystemDeleteHandler();
+                case FILE_ID -> handlers.fileSystemDeleteHandler();
+                default -> throw new UnsupportedOperationException("SystemDelete without IdCase");
+            };
+            case SYSTEM_UNDELETE -> switch (txBody.systemUndeleteOrThrow().id().kind()) {
+                case CONTRACT_ID -> handlers.contractSystemUndeleteHandler();
+                case FILE_ID -> handlers.fileSystemUndeleteHandler();
+                default -> throw new UnsupportedOperationException("SystemUndelete without IdCase");
```
Comment (hendrikebbers) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/dispatcher/TransactionDispatcher.java:189]: please make it consistent if the string is inlined or a constant like for TYPE_NOT_SUPPORTED

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/dispatcher/MonoTransactionDispatcherTest.java [hunk 2]
```diff
@@ -653,13 +658,88 @@ private static Stream<Arguments> getDispatchParameters() {
...
+                        TransactionBody.newBuilder()
+                                .fileCreate(FileCreateTransactionBody.DEFAULT)
+                                .build(),
+                        (Function<TransactionHandlers, TransactionHandler>) TransactionHandlers::fileCreateHandler),
+                Arguments.of(
+                        TransactionBody.newBuilder()
+                                .fileUpdate(FileUpdateTransactionBody.DEFAULT)
+                                .build(),
+                        (Function<TransactionHandlers, TransactionHandler>) TransactionHandlers::fileUpdateHandler),
+                Arguments.of(
```
Comment (hendrikebbers) [inline hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/dispatcher/MonoTransactionDispatcherTest.java:672]: I hate that code.
We should extract a lambda that does all the dirty work and have simple lines that creates the arguments:

```
final BiFunction<Function<TransactionBody.Builder, TransactionBody.Builder>, Function<TransactionHandlers, TransactionHandler>, Arguments> argumentSupplier = (transactionBuilderTransformer, handlerFactory) -> {
    final TransactionBody.Builder builder = TransactionBody.newBuilder();
    final TransactionBody body = transactionBuilderTransformer.apply(builder).build();
    return Arguments.of(body, handlerFactory);
};
argumentSupplier.apply(b -> b.cryptoCreateAccount(CryptoCreateTransactionBody.DEFAULT),
        TransactionHandlers::cryptoCreateHandler);
```

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/dispatcher/TransactionDispatcher.java (hunk 1 above)
Comment (netopyr) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/dispatcher/TransactionDispatcher.java:189]: I am not sure I understand this comment. What needs to be made consistent?

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/dispatcher/MonoTransactionDispatcherTest.java (hunk 2 above)
Comment (netopyr) [reply hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/dispatcher/MonoTransactionDispatcherTest.java:672]: Good idea, will change it.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/dispatcher/TransactionDispatcher.java (hunk 1 above)
Comment (hendrikebbers) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/dispatcher/TransactionDispatcher.java:189]: Sometimes you use a constant for the message of the UnsupportedOperationException and sometimes an inline string.

**Review by hendrikebbers — APPROVED** [review]

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/dispatcher/TransactionDispatcher.java (hunk 1 above)
Comment (netopyr) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/dispatcher/TransactionDispatcher.java:189]: Ah, understood. Will change it in the next PR.

**Pull request merged**